packages/shared/src/components/squads/SquadMemberMenu.tsx

```diff
@@ -1,51 +1,163 @@
 import FlagIcon from '../icons/Flag';
 import { reportSquadMember } from '../../lib/constants';
 import { IconSize } from '../Icon';
+import {
+  SourceMember,
+  SourceMemberRole,
+  SourcePermissions,
+  Squad,
+} from '../../graphql/sources';
+import BlockIcon from '../icons/Block';
+import StarIcon from '../icons/Star';
+import UserIcon from '../icons/User';
+import SquadIcon from '../icons/Squad';
+import { usePrompt } from '../../hooks/usePrompt';
+import { UserShortInfo } from '../profile/UserShortInfo';
+import { ModalSize } from '../modals/common/types';
+import { ContextMenu, ContextMenuItemProps } from '../fields/PortalMenu';
+import { UseSquadActions } from '../../hooks/squads/useSquadActions';
 
-const PortalMenu = dynamic(
-  () => import(/* webpackChunkName: "portalMenu" */ '../fields/PortalMenu'),
-  {
-    ssr: false,
-  },
-);
+interface SquadMemberMenuProps
+  extends Pick<UseSquadActions, 'onUpdateRole' | 'verifyPermission'> {
+  squad: Squad;
+  member: SourceMember;
+}
+
+enum MenuItemTitle {
+  AddAsOwner = 'Add as co-owner',
+  PromoteToModerator = 'Promote to moderator',
+  DemoteToModerator = 'Demote to moderator',
+  DemoteToMember = 'Demote to member',
+  RemoveMember = 'Remove member',
+}
+
+const promptDescription: Record<
+  MenuItemTitle,
+  (memberName: string, squadName: string) => string
+> = {
+  [MenuItemTitle.AddAsOwner]: (memberName, squadName) =>
+    `${memberName} will get the same permissions as you have. Adding as co-owner will not replace you as the owner of ${squadName}.`,
+  [MenuItemTitle.PromoteToModerator]: (memberName) =>
+    `${memberName} will now get moderator permissions and be able to remove posts and members form your Squad. You can reverse this decision later.`,
+  [MenuItemTitle.DemoteToModerator]: (memberName) =>
+    `${memberName} will no longer have owner permissions. You can always reverse this decision later.`,
+  [MenuItemTitle.DemoteToMember]: (memberName) =>
+    `${memberName} will lose the moderator permissions and become a regular member.`,
+  [MenuItemTitle.RemoveMember]: (memberName, squadName) =>
+    `${memberName} will be a Blocked members and will no longer have access to ${squadName}. They will not be able to rejoin unless you unblock them.`,
+};
 
 export default function SquadMemberMenu({
-  squadId,
-  memberId,
-}: {
-  squadId: string;
-  memberId: string;
-}): ReactElement {
+  squad,
+  member,
+  onUpdateRole,
+  verifyPermission,
+}: SquadMemberMenuProps): ReactElement {
   const { user } = useContext(AuthContext);
+  const { showPrompt } = usePrompt();
+  const onUpdateMember = async (
+    role: SourceMemberRole,
+    title: MenuItemTitle,
+  ) => {
+    const hasConfirmed = await showPrompt({
+      title: `${title}?`,
+      description: promptDescription[title](member.user.name, squad.name),
+      okButton: { title, className: 'btn-primary-ketchup' },
+      content: <UserShortInfo user={member.user} />,
```

Comment:
The remove role currently doesn't show this, is there any way we can dynamic map this as well?

Reply:
Good catch! Coming right up!

Reply:
After checking with the product/design team, they now want to keep the preview all the time.

Quick link to discussion: https://dailydotdev.slack.com/archives/C04UQUBDWKD/p1680526000089179?thread_ts=1680522693.874399&cid=C04UQUBDWKD